src/pages/en/guides/migration-guide/from-gatsby.md

```diff
@@ -0,0 +1,1199 @@
+---
+title: Migrating from Gatsby
+description: Tips for migrating an existing Gatsby project to Astro
+layout: ~/layouts/CMSLayout.astro
+stub: false
+framework: Gatsby
+---
+
+Here are some tips for converting a Gatsby project to Astro. This is not a full, step-by-step walkthrough, but it will guide you through some changes you will have to make. 
+
+## Key Similarities
+
+While we'll touch on the differences between Gatsby and Astro shortly, it's important to note the ways in which the two overlap.
+
+- The syntax of `.astro` files is similar to JSX. Writing Astro should feel familiar.
+
+- Astro is component-based. Your markup structure will likely align closely before and after your migration.
+
+- Astro has built-in support for Markdown and an integration for using MDX files. Both Gatsby and Astro use [Remark](https://remark.js.org/) by default for Markdown manipulation and pre-processing, so you can continue to use many of your existing Remark plugins.
+
+- Astro also has [official integrations for using React components](/en/guides/integrations-guide/react/). Note that in Astro, React files **must** have a `.jsx` or `.tsx` extension.
+
+- Astro projects can also be SSG or SSR. (Support for per-page rendering strategy is planned.)
+
+- Astro has support for NPM package usage, including several for React. You may be able to keep some or all of your existing React components and dependencies.
+
+## Key Differences
+
+Now that you understand the ways that Astro and Gatsby align, how do they diverge?
+
+### React App vs MPA 
+
+Gatsby is a React app, and uses `index.js` as your project's root. While Gatsby is able to generate an HTML page for each of the routes configured, it also re-initializes React for all of the content on-screen, including static portions.
+
+Astro is a multi-page site, and `index.astro` is your home page. It generates HTML pages for each configured route but, in contrast to Gatsby, only initializes JavaScript on the interactive elements on-screen.
+
+### Page routing
+
+Because of the differences between React apps and Astro MPA apps, you don't need to know any domain-specific knowledge to link between different pages. Instead of Gatsby's custom `<Link to="/path">` component, you'll use the HTML standard `<a href="/path">` tag.
+
+In addition, while Gatsby primarily uses the `createPages` function to create dynamic pages, [Astro opts for page-based routing](/en/core-concepts/routing/#dynamic-routes) that better represents the individual HTML files that are output in a build.
+
+Finally, should you need to access information about the current route within an Astro component, [you're able to use `Astro.url`](/en/reference/api-reference/#astrourl), which exposes a [web-standard `URL` class](https://developer.mozilla.org/en-US/docs/Web/API/URL).
```

Comment:
The following suggestion was originally written by @sarah11918:

```suggestion
### Page routing

You don't need to know any domain-specific knowledge to link between different pages in Astro. Instead of Gatsby's custom `<Link to="/path">` component, use the HTML standard `<a href="/path">` tag.

In addition, while Gatsby primarily uses the `createPages` function to create dynamic pages, [Astro opts for page-based routing](/en/core-concepts/routing/#dynamic-routes) that better represents the individual HTML files that are output in a build.

To access information about the current route, use [`Astro.url`](/en/reference/api-reference/#astrourl) to expose a [web-standard `URL` class](https://developer.mozilla.org/en-US/docs/Web/API/URL).
```

Fixed the links to make them relative inside the project, and made the first and third points a little more direct. (Use/ vs. you'll use; not annotating "additionally/finally")

I grok the middle point, but I think the wording needs some work. The points to me are:

- Gatsby needs `createPages` AT ALL to... create any pages! Astro CAN create pages (non-dynamically) on file-placement alone. Dynamic page routes are not in fact necessary (although, totally possible!)
- `createPages` is like a config setup thing to create ALL your pages. Astro's `getStaticPaths` is right inside `src/pages/`, collocated with the set of pages you want to build dynamically. Maybe that's the way in which Astro's dynamic page routing is "closer to the platform of HTML"?

Is there any other nuance about that I'm not getting that should be represented? Can we keep working on the wording of point 2?